examples/ecs/observers.rs

```diff
@@ -0,0 +1,206 @@
+//! Demonstrates how to observe life-cycle triggers as well as define custom ones.
+
+use bevy::{
+    ecs::observer::Observer,
+    prelude::*,
+    utils::{HashMap, HashSet},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<SpatialIndex>()
+        .add_systems(Startup, setup)
+        .add_systems(Update, (draw_shapes, handle_click))
+        // Observers are systems that run when an event is "triggered". This observer runs whenever
+        // `ExplodeMines` is triggered.
+        .observe(
+            |trigger: Trigger<ExplodeMines>,
+             mines: Query<&Mine>,
+             index: Res<SpatialIndex>,
+             mut commands: Commands| {
+                // You can access the trigger data via the `Observer`
+                let event = trigger.event();
+                // Access resources
+                for e in index.get_nearby(event.pos) {
+                    // Run queries
+                    let mine = mines.get(e).unwrap();
+                    if mine.pos.distance(event.pos) < mine.size + event.radius {
+                        // And queue commands, including triggering additional events
+                        // Here we trigger the `Explode` event for entity `e`
+                        commands.trigger_targets(Explode, e);
+                    }
+                }
+            },
+        )
+        // This observer runs whenever the `Mine` component is added to an entity, and places it in a simple spatial index.
+        .observe(add_mine)
+        // This observer runs whenever the `Mine` component is removed from an entity (including despawning it)
+        // and removes it from the spatial index.
+        .observe(remove_mine)
```

Comment:
I don't want to bikeshed naming, but I find the names `add_mine` and `remove_mine` slightly confusing. Without reading the comment above them, it's ambiguous whether the system's purpose is to add mines, or react to mines being added.

Typically, system names are verbs describing _what_ the systems do. A system named `setup_paddle` would be responsible for setting up a paddle entity. I don't want to introduce an ambiguity where we have to wonder if `setup_paddle` might really be an observer on a `Trigger<OnSetup, Paddle>`.

Not sure what a good naming convention should be, if any. `on_add_mine` or `handle_mine_add` both convey that these are observer systems, but break the convention of describing what the systems do. On the other hand, `add_mine_to_spatial_index` immediately conveys what the system does, while being less explicit that it's an observer system.

Thoughts?

Reply:
I agree with this complaint :) IMO `handle` or `respond` are the right verbs here.

Reply:
Alternatively, we could swap the verbs to past tense: `mine_added` and `mine_removed`.

Reply:
The Event is already the verb here imo. This should be on_add_mine imo.